src/foam/nanos/crunch/ruler/SetUCJStatusOnPut.js

```diff
@@ -63,11 +63,17 @@ foam.CLASS({
               ucj.setStatus(CapabilityJunctionStatus.APPROVED);
             } else if ( chainedStatus == CapabilityJunctionStatus.GRANTED && reviewRequired ) {
               ucj.setStatus(wasApproved ? CapabilityJunctionStatus.GRANTED : CapabilityJunctionStatus.PENDING);
+
+            // Change a previously saved onboarding remains ACTION_REQUIRED to
+            // a GRANTED state after filling out the rest of the information in
+            // a wizard and submitting it
+            } else if ( chainedStatus == CapabilityJunctionStatus.ACTION_REQUIRED && reviewRequired ) {
```

Comment:
So what capability exactly is remaining in ACTION_REQUIRED in the cardsOnboarding flow. Think this is more specific to the cardsRequirements capability chain rather than SetUCJStatusOnPut. It doesn't seem like we should be auto setting ucj with ACTION_REQUIRED and reviewRequired to GRANTED. If you find the UCJ that gets stuck in action-required it may be more of a problem with that rule not retriggering instead.

Reply:
To @tharmaman's point, a capability coming into this rule as `ACTION_REQUIRED` is considered to have not been validated. If it's really ready to go to `GRANTED` it should come into this rule with `PENDING` status.